### PR TITLE
Pin transformers version <4.47

### PR DESCRIPTION
**Pull request by kashif**
# What does this PR do?
pins the transformers version to <4.47

fixes [#2445](https://github.com/huggingface/trl/issues/2445)

### CODE REVIEW

**Review by lewtun — APPROVED** [review]
LGTM, don't forget to add a tag after merging this PR

**Comment by anhuong** [comment]
Does this also need to be updated in the [requirements.txt](https://github.com/huggingface/trl/blob/main/requirements.txt#L4)? as it still shows `transformers>=4.46.0`

**Comment by kashif** [comment]
@anhuong I believe the `requirements.txt` is used by the CI and the issue is fixed in the main branch...

**Pull request merged**
